### ISSUE TITLE
[atlmfc:x64-windows-static] build failure

### ISSUE THREAD

**Issue by GabrielDougherty** [microsoft/vcpkg#31766]
### Operating system

Windows 10

### Compiler

MSVC 2019

### Steps to reproduce the behavior

Open Developer Command Prompt for VS 2019

```Shell

cd c:\dev\vcpkg
.\vcpkg install boost-circular-buffer:x64-windows-static boost-stacktrace:x64-windows-static openssl:x64-windows-static qt5-base:x64-windows-static qt5-svg:x64-windows-static qt5-tools:x64-windows-static qt5-winextras:x64-windows-static

```

### Failure logs

```
Additional packages (*) will be modified to complete this operation.
boost-circular-buffer:x64-windows-static is already installed
Detecting compiler hash for triplet x64-windows-static...
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\Gabriel Dougherty\AppData\Local\vcpkg\archives in 1.35 ms. Use --debug to see more details.
Installing 1/54 atlmfc:x64-windows-static...
Building atlmfc[core]:x64-windows-static...
CMake Error at ports/atlmfc/portfile.cmake:7 (message):
  Unable to locate 'afxres.h'.  Ensure you have installed the ATL/MFC
  component of Visual Studio.
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)


error: building atlmfc:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+atlmfc
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[atlmfc]+Build+error&body=Copy+issue+body+from+C%3A%2Fdev%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```

### Additional context

I have MFC/ATL installed for Visual Studio 2019.

![image](https://github.com/microsoft/vcpkg/assets/10541219/ec6dc44f-7c76-4022-a818-e849509c7d1f)

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. Please rebuild `atlmfc` by command `.\vcpkg install atlmfc:x64-windows-static --debug`. And please provide the new log.

**Comment by GabrielDougherty** [comment]
Ok, here is the log: https://gist.github.com/GabrielDougherty/a615d25622da547572a3977607b08012

**Comment by LilyWangLL** [comment]
Your log show you are using the header file from Visual Studio 2022. Did you install ATL and MFC component on Visual Studio 2022?
```
EXTERNAL_INCLUDE=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\ATLMFC\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um
Framework40Version=v4.0
FrameworkDir=C:\Windows\Microsoft.NET\Framework64\
FrameworkDir64=C:\Windows\Microsoft.NET\Framework64\
FrameworkVersion=v4.0.30319
FrameworkVersion64=v4.0.30319
FSHARPINSTALLDIR=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\FSharp\Tools
HOMEDRIVE=C:
HOMEPATH=\Users\Gabriel Dougherty
INCLUDE=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\ATLMFC\include;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um
```

**Comment by GabrielDougherty** [comment]
Yes, why is it using Visual Studio 2022 includes when I am using the VS2019 command prompt? Sorry I am new new Windows C++ development. I wanted to try it out after developing C++ on Linux professionally for 4+ years.

**Comment by LilyWangLL** [comment]
If you want to use a specific version of Visual Studio, please add `set(VCPKG_PLATFORM_TOOLSET v142)` in the triplets file. The detection of Visual Studio is done by the vcpkg.exe itself, and the default strategy is to use the latest version of Visual Studio, unaffected by the command prompt.

**Comment by GabrielDougherty** [comment]
@LilyWangLL thank you. That fixed my issue. In my case I had to edit `C:\dev\vcpkg\triplets\x64-windows-static.cmake` and add that line to the file. I found this helpful blog post: https://devblogs.microsoft.com/cppblog/vcpkg-using-multiple-enlistments/

It appears that if I want to use libraries that depend on ATL/MFC on different visual studio versions, I will need to maintain separate vcpkg roots, called "enlistments".

@sfhacker stop leaving "thumbs down" reacts on folks' posts. It is unkind and rude. I also noticed across many open source projects that when you do leave a comment, it is often condescending or unhelpful. Also the way you criticize Chinese people's grammar on GitHub appears to be intentionally unkind and rude. I've raised a report with GitHub on your conduct.
